loopchain/peer/block_manager.py

```diff
@@ -298,7 +298,7 @@ def add_confirmed_block(self, confirmed_block: Block, confirm_info=None):
             if result:
                 return
 
-        self.block_height_sync()
+        self.__channel_service.state_machine.block_sync()
```

Comment:
How about raising an exception instead of block sync.
Determining if it needs to sync blocks is placed out of this method.

Reply:
This line is removed, expecting channel_service would handle the exception during publish_new_block in node_subscriber.